pkg/didcomm/protocol/didexchange/persistence.go

```diff
@@ -86,14 +99,60 @@ func (c *ConnectionRecorder) GetInvitation(verKey string) (*Invitation, error) {
 	return result, nil
 }
 
-// invitationKey computes key for invitation object
-func invitationKey(verKey string) (string, error) {
-	storeKey, err := computeHash([]byte(verKey))
+// SaveConnectionRecorder saved the connection record against the connection id  in the store
+func (c *ConnectionRecorder) SaveConnectionRecorder(invitation *Invitation,
+	msg *Message) error {
+	connRecord := &connectionRecord{ThreadID: msg.ThreadID,
+		ServiceEndPoint: invitation.ServiceEndpoint,
+		RecKey:          invitation.RecipientKeys, State: msg.State}
+
+	connBytes, err := json.Marshal(connRecord)
+	if err != nil {
+		return fmt.Errorf("error while marshalling save connection record %s : ", err)
+	}
+	return c.store.Put(msg.ConnectionID, connBytes)
+}
+
+// SaveProtocolID saves the hash of the protocol id which is a did and thread id as an ID and connection id as value.
+func (c *ConnectionRecorder) SaveProtocolID(myDid, thid, connectionID string) error {
+	protocolID, err := computeHash([]byte(myDid + thid))
+	if err != nil {
+		return fmt.Errorf("error while computing hash : %s", err)
+	}
+	return c.store.Put(protocolID, []byte(connectionID))
+}
+
+// GetConnectionID returns the connection ID based on hashed protocol ID
+func (c *ConnectionRecorder) GetConnectionID(myDID, thid string) (string, error) {
+	protocolID, err := computeHash([]byte(myDID + thid))
 	if err != nil {
 		return "", err
 	}
+	connectionIDBytes, err := c.store.Get(protocolID)
```

Comment:
use prefix here and other places in this file

Reply:
done